kolibri/content/utils/channel_import.py

```diff
@@ -222,20 +239,72 @@ def table_import(self, model, row_mapper, table_mapper, unflushed_rows):
             self.destination.session.bulk_insert_mappings(DestinationRecord, data_to_insert)
         return unflushed_rows
 
+    def merge_record(self, data, model, DestinationRecord):
+        # Models that should be merged (see list above) need to be individually merged into the session
+        # as SQL Alchemy ORM does not support INSERT ... ON DUPLICATE KEY UPDATE style queries,
+        # as not available in SQLite, only MySQL as far as I can tell:
+        # http://hackthology.com/how-to-compile-mysqls-on-duplicate-key-update-in-sql-alchemy.html
+        RowEntry = self.destination.session.query(DestinationRecord).get(data[model._meta.pk.name])
+        if RowEntry:
+            for key, value in data.items():
+                setattr(RowEntry, key, value)
+        else:
+            RowEntry = DestinationRecord(**data)
+        self.destination.session.merge(RowEntry)
+
+    def check_and_delete_existing_channel(self):
```

Comment:
Why do we need to delete a channel in import?

Reply:
We only reimport when the channel version has changed now, and we can't guarantee that nodes that previously existed exist in the new version.

We could try to do a diff to do a more selective update, but this is simpler and produces the same result.